generator/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/generator/federation/validation/validateRequiresDirective.kt

```diff
@@ -20,14 +20,11 @@ import com.expediagroup.graphql.generator.federation.directives.REQUIRES_DIRECTI
 import graphql.schema.GraphQLFieldDefinition
 
 // [OK]    @requires references valid fields marked @external
-// [ERROR] @requires specified on base type
+// [OK]    @requires specified on base type
 // [ERROR] @requires specifies non-existent fields
 internal fun validateRequiresDirective(validatedType: String, validatedField: GraphQLFieldDefinition, fieldMap: Map<String, GraphQLFieldDefinition>, extendedType: Boolean): List<String> {
-    val errors = mutableListOf<String>()
     if (extendedType) {
-        errors.addAll(validateDirective("$validatedType.${validatedField.name}", REQUIRES_DIRECTIVE_NAME, validatedField.allAppliedDirectivesByName, fieldMap, extendedType))
-    } else {
-        errors.add("base $validatedType type has fields marked with @requires directive, validatedField=${validatedField.name}")
+        return validateDirective("$validatedType.${validatedField.name}", REQUIRES_DIRECTIVE_NAME, validatedField.allAppliedDirectivesByName, fieldMap, extendedType)
```

Comment:
this validation should be applied for both base and extended type -> it doesn't make sense to keep this file anymore, move this `validateDirective` call directly to [`FederatedSchemaValidator`](https://github.com/ExpediaGroup/graphql-kotlin/blob/82ae4bb4de0d26658dc16ed950a9d4f91362b788/generator/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/generator/federation/validation/FederatedSchemaValidator.kt#L71)